packages/node/src/machine/enums.ts

```diff
@@ -23,7 +23,14 @@ enum Opcode {
   /**
    * Middleware hook to both send and wait for a response from a ProtocolMessage
    */
-  IO_SEND_AND_WAIT
+  IO_SEND_AND_WAIT,
+
+  /**
+   * Middleware hook to check for valid state proposal if a relevant registered
+   * plugin is available. Otherwise the functionality of this hook is
+   * auto-signing the proposed state.
+   */
+  OP_VALIDATE_STATE_PROPOSAL
```

Comment:
Can't we register a middleware hook on `OP_SIGN`? Why are we adding a new opcode?

Reply:
We could but there's no specification on the ordering of the middlewares that get run when registered against a hook.

So we could end up signing an update that the plugin actually rejects.

Reply:
@IIIIllllIIIIllllIIIIllllIIIIllllIIIIll thoughts?

Reply:
This thing should only be placed in `update` right? We don't want to hook into the `OP_SIGN`s in the other protocols